src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs

```diff
@@ -29,10 +29,16 @@ internal TarEntry(TarHeader header, TarReader readerOfOrigin, TarEntryFormat for
         }
 
         // Constructor called when the user creates a TarEntry instance from scratch.
-        internal TarEntry(TarEntryType entryType, string entryName, TarEntryFormat format)
+        internal TarEntry(TarEntryType entryType, string entryName, TarEntryFormat format, bool isGea)
         {
             ArgumentException.ThrowIfNullOrEmpty(entryName);
-            TarHelpers.ThrowIfEntryTypeNotSupported(entryType, format);
+
+            Debug.Assert(!isGea || (isGea && entryType is TarEntryType.GlobalExtendedAttributes));
+
+            if (!isGea)
+            {
+                TarHelpers.ThrowIfEntryTypeNotSupported(entryType, format);
+            }
```

Comment:
It feels odd to me to add a ctor parameter to track this one format specially.  But I guess the alternative is to move the ThrowIf call to every ctor other than PAX GEA, and that's not necessarily better when there's only one exception.

Reply:
Yeah you deduced right. I had many options to throw when the entry was PAX, and I thought this method would be the least intrusive of all.